Tests/Get-BMEnvironment.Tests.ps1

```diff
@@ -0,0 +1,181 @@
+{
+    $environments | Should -BeNullOrEmpty
+}
+
+function ThenEnvironmentsReturned
+{
+    param(
+        [Parameter(Mandatory)]
+        [string[]]$Named,
+
+        [Switch]$AndInactiveEnvironments
+    )
+
+    if( $AndInactiveEnvironments )
+    {
+        $environments | Should -Not -BeNullOrEmpty
+    }
+    else
+    {
+        $environments | Should -HaveCount $Named.Count
+    }
+    foreach( $name in $Named )
+    {
+        $environments | Where-Object { $_.Name -eq $name } | Should -Not -BeNullOrEmpty
+    }
+}
+
+function WhenGettingEnvironments
+{
+    [CmdletBinding()]
+    param(
+        [string]$Named,
+
+        [Switch]$Force
+    )
+
+    $optionalParams = @{ }
+    if( $Named )
+    {
+        $optionalParams['Name'] = $Named
+    }
+    if( $Force )
+    {
+        $optionalParams['Force'] = $true
+    }
+    $script:environments = Get-BMEnvironment -Session $session @optionalParams
+}
+
+Describe 'Get-BMEnvironment.when given no name' {
+    It ('should return all active server environments') {
+        Init
+        GivenEnvironment 'One'
+        GivenEnvironment 'Two'
+        WhenGettingEnvironments
+        ThenEnvironmentsReturned 'One','Two'
+        ThenNoErrorWritten
+    }
+}
+
+Describe 'Get-BMEnvironment.when given no name and using the Force' {
+    It ('should return all active and inactive server environments') {
+        Init
+        GivenEnvironment 'One'
+        WhenGettingEnvironments -Force
+        ThenEnvironmentsReturned 'One' -AndInactiveEnvironments
+        ThenNoErrorWritten
+    }
+}
+
+Describe 'Get-BMEnvironment.when given name to an active environment' {
+    It ('should return named environment') {
+        Init
+        GivenEnvironment 'One'
+        GivenEnvironment 'Two'
+        WhenGettingEnvironments -Named 'One'
+        ThenEnvironmentsReturned 'One'
+    }
+}
+
+Describe 'Get-BMEnvironment.when given name to an inactive environment' {
+    It ('should return named environment') {
+        Init
+        GivenEnvironment 'One' -Disabled
+        GivenEnvironment 'Two'
+        WhenGettingEnvironments -Named 'One'
+        ThenEnvironmentsReturned 'One'
+    }
+}
+
+Describe 'Get-BMEnvironment.when given wildcards' {
+    It ('should return only active environments whose name match the wildcard') {
+        Init
+        GivenEnvironment 'One'
+        GivenEnvironment 'Onf'
+        GivenEnvironment 'Ong' -Disabled
+        GivenEnvironment 'Two'
+        WhenGettingEnvironments -Named 'On*'
+        ThenEnvironmentsReturned 'One','Onf'
+    }
+}
+
+Describe 'Get-BMEnvironment.when given wildcards and using the Force' {
+    It ('should return active and inactive environments whose name match the wildcard') {
+        Init
+        GivenEnvironment 'One'
+        GivenEnvironment 'Onf'
+        GivenEnvironment 'Ong' -Disabled
+        GivenEnvironment 'Two'
+        WhenGettingEnvironments -Named 'On*'
+        ThenEnvironmentsReturned 'One','Onf'

```

Comment:
Test is incorrect, this assertion should be `ThenEnvironmentsReturned 'One', 'Onf', 'Ong'`